### PR TITLE
Add `udx_send_ttl()` and `ttl` argument in `Socket.send()`

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by mafintosh** [comment]
Looks great as is, anything missing?

**Comment by kasperisager** [comment]
Just the JavaScript side.

**Comment by mafintosh** [comment]
Nice, that's just as simple as calling the right method yea?

File: lib/socket.js [hunk 1]
```diff
@@ -142,17 +142,26 @@ module.exports = class UDX extends events.EventEmitter {
...
   }
 
-  send (buffer, offset, length, port, ip, onflush) {
+  send (buffer, offset, length, port, ip, ttl, onflush) {
     if (this.closing) throw new Error('Socket is closed')
     if (!this.bound) this.bind(0)
 
+    if (typeof ttl === 'function') {
+      onflush = ttl
+      ttl = undefined
```
Comment (mafintosh) [inline lib/socket.js:151]: ttl = 0

Comment (mafintosh) [reply lib/socket.js:151]: (ie non polymorphic)

File: lib/socket.js [hunk 2]
```diff
@@ -142,17 +142,26 @@ module.exports = class UDX extends events.EventEmitter {
...
 
     req.buffer = buffer.subarray(offset, offset + length)
     req.onflush = onflush || noop
 
-    binding.udx_napi_send(this._handle, req.handle, id, buffer, port, ip)
+    if (ttl) {
+      binding.udx_napi_send_ttl(this._handle, req.handle, id, buffer, port, ip, ttl)
+    } else {
+      binding.udx_napi_send(this._handle, req.handle, id, buffer, port, ip)
+    }
```
Comment (mafintosh) [inline lib/socket.js:164]: Isn't this equi? 

``` js
binding.udx_napi_send_ttl(this._handle, req.handle, id, buffer, port, ip, ttl || 0)
```

Then we can remove the normal send method from the bindings also

Comment (kasperisager) [reply lib/socket.js]: A TTL of 0 will be rejected with `EINVAL` so I've made the binding switch to `udx_send()` when TTL is 0.

File: binding.c [hunk 3]
```diff
@@ -330,7 +331,12 @@ NAPI_METHOD(udx_napi_send) {
 
   uv_buf_t b = uv_buf_init(buf, buf_len);
 
-  err = udx_send(req, self, &b, 1, (const struct sockaddr *) &addr, on_udx_send);
+  if (ttl == 0) {
+    err = udx_send(req, self, &b, 1, (const struct sockaddr *) &addr, on_udx_send);
+  } else {
+    err = udx_send_ttl(req, self, &b, 1, (const struct sockaddr *) &addr, ttl, on_udx_send);
+  }
+
```
Comment (mafintosh) [inline binding.c:339]: this if isn't needed now is it? just call _ttl always?

Comment (kasperisager) [reply binding.c:339]: For reference:

https://github.com/mafintosh/libudx/blob/b90232ba01557bee396274294b641f45fca43e7d/src/udx.c#L774

We could specifically allow `ttl == 0`.

Comment (mafintosh) [reply binding.c:339]: Yea I think we should allow 0 also

**Comment by mafintosh** [comment]
👍

**Pull request merged**